### ISSUE TITLE
Inconsistent loading order of CSS Modules + global CSS

### ISSUE THREAD

**Issue by nikkiselev** [zeit/next.js#10148]
# Bug report

## Describe the bug

I have a global CSS imported in `/pages/_app.js` and `name.module.css` imported in a page. In production mode the CSS Module is loaded **after** the global CSS. In development mode, the CSS Module is loaded **before** the global CSS.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

[Repository with code needed to reproduce the issue](https://github.com/nikkiselev/nextjs-css)

1. [Add  a global style within pages/_app.js](https://nextjs.org/docs/basic-features/built-in-css-support#adding-a-global-stylesheet)
2. [Add and import a CSS Module](https://nextjs.org/docs/basic-features/built-in-css-support#adding-component-level-css)
3. Add CSS classes to both files with the same specificity, so the order of including files resolves the conflict.
3. Run "dev" script.
4. See order of CSS in a browser.
5. Run "build" and "start" scripts.
6. See different order of CSS in a browser.

## Expected behavior

In development mode the global CSS should be loaded first and CSS Modules later on, the same as it works in production.

## System information

- OS: macOS 10.15.2
- Browser: Chrome 79.0.3945.130
- Version of Next.js: 9.2.0

## Additional context

The page looks different in dev and production modes because of CSS overriding.

**Comment by Timer** [comment]
Please provide an example of the CSS files used.

**Comment by nikkiselev** [comment]
I've added a repository with minimum code that reproduces the issue https://github.com/nikkiselev/nextjs-css

**Comment by liborvanek** [comment]
Once this issue is resolved, will there be any way to control the order of imported CSS files?
Looks like the default order is global CSS first, CSS modules afterwards. 
Frem previous comments I understood it wasn't possible to modify CSS import order when using `next-css`, but it might be possible using new `CSS Support`. But I can't seem to find any mention neither in the RFC, docs nor release notes.

A common desirable pattern might look like this:
```
– resets
– global CSS (typography etc.)
– CSS modules
– global utilities
```

**Comment by timneutkens** [comment]
CSS resets are global CSS, you can import them first in _app.js and it'll be loaded in that order.

```
// in pages/_app.js
import 'reset.css'
import 'global.css'
```

**Comment by liborvanek** [comment]
@timneutkens Yes, I know. Sorry for not being clear. My question is more like: Are we able to import CSS, that would be loaded **after** CSS modules?

**Comment by timneutkens** [comment]
You can't know the load order of CSS modules as they get added per-page. So there really is no "after" CSS modules state.

**Comment by liborvanek** [comment]
@timneutkens I don't need to know the load order of CSS modules, I just need to find a way to include a global CSS **at the very end of** `<head>`, so dynamically loaded CSS modules are injected always **before** this stylesheet, not after.
Does that make sense? Is it even possible?

**Comment by Timer** [comment]
@liborvanek that is not possible, nor a use case we're currently interested in exploring. CSS will be loaded the order it's imported by the application!

**Comment by Exelord** [comment]
Any update on this issue? It's blocking by styles development using sass :/

**Comment by pav-workchain** [comment]
What's the progress with this? I can't use the latest version on next because of this. I load two stylesheets in my project, bootstrap first then my styles. When I deploy to production the load order is reversed and all my styles are overwritten by bootstrap's...

**Comment by bravokiloecho** [comment]
Shouldn't any CSS imported from CSS modules override global styles  imported in the `_app.js` file?

If I land on a page with a component that imports a CSS module, the global styles are added below the module's styles and overwrite it's rules. But if I start the app _and then_ go to the page with the component, the component's CSS gets added under the global styles and overwrites them.

I would have thought the second version is the ideal behaviour. But it definitely shouldn't be inconsistent no?

**Comment by valse** [comment]
We have the same issue using the version 9.3.0 with SASS globals in _app and the CSS modules for single components.

**Comment by jahirfiquitiva** [comment]
I'm currently facing the same issue and it's kinda frustrating. It's stopping me from using next for now. 😕

**Comment by jrobchin** [comment]
Anyone have any ideas on how to resolve this? It's making it difficult to use Next as intended.

**Comment by sparkertime** [comment]
I too am suffering from this. Right now my only solution is to switch to cycles of `yarn build && yarn start` when working on CSS. This is also fixed by reverting to 9.1.7 and using the next-css / next-sass plugins, but that leaves a lot of great recent features on the table. I was hoping I could leverage the `composes` feature of css-modules to work around this but I think that's blocked by a separate issue in https://github.com/zeit/next.js/issues/10142

I took a look at the CSS loading code in next as well as what magic happens in `yarn build` and it's beyond me, but I really hope this gets a PR soon.

**Comment by sparkertime** [comment]
If anyone on this thread is desperate for this fix, it is available in the `9.3.6-canary.0` next.js release (thanks @Timer for turning that around so quickly!)

**Comment by jrobchin** [comment]
@sparkertime this is awesome, thanks!

**Comment by Joozty** [comment]
@Timer many thanks for looking into it. Nonetheless, it's still broken when I import some components in `_app.jsx`. I prepared a repo to reproduce. Hope it helps :) 

https://github.com/Joozty/nextjs-css

**Comment by sparkertime** [comment]
@Joozty You'll probably want to open an issue specifically for ordering when importing modules from `_app.jsx`. This issue is now closed and what you're reporting is definitely beyond my ability to fix, so hopefully a new issue will get some attention from the core team

**Comment by chornos13** [comment]
This is still not fixed yet, I still experienced with this issue in production mode in v9.4.4

**Comment by chornos13** [comment]
after finding out what is the problem, I found that this issue because of mini-css plugin, I tried with my own version of webpack-config, after I added mini-css plugin to build on production this bug happened.

**Comment by mmiszy** [comment]
9.4.4 did NOT fix the problem for me. However, @canary (currently 9.4.5-canary.7) DOES fix it in my case.
Hope that helps.

**Comment by qiwang97** [comment]
> This is still not fixed yet, I still experienced with this issue in production mode in v9.4.4,
> I use custom CSS loader and use antd-design

same problem. I use reset.scss in _app.tsx and it does not work. Any  solution now?

**Comment by chshanovskiy** [comment]
I faced this problem too.

In my _app.tsx, I imported single css file, which `@import`ed other global styles. And this led to a problem. 
The solution was to import ALL global css files inside the _app.tsx.

Before:
```
// _app.tsx
import "../styles/index.css"

// styles/index.css
@import "./reset.css";
@import "./grid.css";
```

After:
```
// _app.tsx
import "../styles/reset.css";
import "../styles/grid.css";
```

**Comment by hydRAnger** [comment]
This problem is still there, is there any plan on fixing it?

**Comment by ysthedood** [comment]
Not really a fix, but a temporary solution is to directly copy and paste the files into one document to prevent Next from re-ordering them. 

I am on Next v9.5.2. This is preventing me from using next for future projects. Could you please advise when this will be fixed? Or is there a way to specify the order of imports for css and scss files?

**Comment by Timer** [comment]
I'm locking this issue because the original issue was solved. If you are experiencing a bug, please open a new issue.
